backend/django/create-model-and-api.mo.md

```diff
@@ -0,0 +1,219 @@
+
+## Steps  
+
+- In this tutorial we will create a News model, add it to the Django admin back-office and an API route to get all the pieces of news that have been created.
+- To give a full example, we will ==== REMPLIR
+
+### Create a News model *(~10 min)*
+
+- In this example we will create a new model called News, with different attributes as you will see bellow. In the models.py file of your app (in our case Publications) create a new model. 
+
+ourproject/publications/models.py
+```python
+from django.db import models
+
+class News(models.Model):
+    title = models.CharField(max_length=50)   #CharField is a single line form in the admin; max_length speaks for itself
+    description = models.TextField(max_length=1000)   #TextField is a scrollable paragraph in the admin
+    created_at = models.DateTimeField(auto_now_add=True)    #auto_now_add generates automatically a timestamp when you create your create the news
+    updated_at = models.DateTimeField(auto_now=True)    #auto_now updates the timestamp everytime you change the news
+    city = models.ForeignKey('locations.city')    #All the cities of the City model of the locations app
+
+    class Meta:
+        verbose_name_plural = "News"    #The name of the class should be singular, this forces the plural to be News and not Newss
+
+    def __str__(self):
+        return self.title
+```
+
+
+- If you have users that are in groups, they need the permission to access to the Model. We won't explain how to create a group, we just want to show you how to add permissions to them. They will be added automaticcaly when you deploy.
+
+ouproject/users/apps/py
+```python
+from django.apps import AppConfig
+from django.contrib.admin import site
+from django.db.models.signals import post_migrate
+
+
+class UsersConfig(AppConfig):
+    name = 'intramuros.users'
+    verbose_name = "Users"
+
+    def ready(self):
+        post_migrate.connect(add_group_permissions, sender=self)
+
+
+def add_model_permissions(group, model, ContentType, Permission):
+    content_type = ContentType.objects.get_for_model(model)
+    permissions = Permission.objects.filter(content_type=content_type)
+    for permission in permissions:
+        group.permissions.add(permission)
+
+
+def add_group_permissions(sender, using, apps, **kwargs):
+    Group = apps.get_model("auth", "Group")
+    Permission = apps.get_model("auth", "Permission")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+    News = apps.get_model("publications", "News")   #Add this line
+
+    if Group.objects.using(using).filter(name='Mayor Admin').exists():
+        group = Group.objects.get(name='Mayor Admin')
+    else:
+        group = Group.objects.using(using).create(name='Mayor Admin')
+    add_model_permissions(group, News, ContentType, Permission) #Add this second line
+```
+
+- Finally, make your migrations and migrate.
```

Comment:
How do you do that?

Reply:
How do you migrate down if you forgot something?

Reply:
YEs explain how to migrate down to a former migration. Simply set the name of the migration './manage.py migrate app_name 0005_the_old_migration'.

Reply:
If you forgot a field, or it was not set correctly you can [squash migrations](https://docs.djangoproject.com/fr/1.11/ref/django-admin/#squashmigrations)